src/ios/LocalyticsPlugin.m

```diff
@@ -942,29 +1124,38 @@ - (void)getInAppMessageDismissButtonLocation:(CDVInvokedUrlCommand *)command {
 }
 
 - (void)triggerInAppMessage:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"triggerInAppMessage" withCommand:command];
     NSString *triggerName = [command argumentAtIndex:0];
     NSDictionary *attributes = [command argumentAtIndex:1];
 
     if ([triggerName isKindOfClass:[NSString class]] && [triggerName length] > 0) {
         [Localytics triggerInAppMessage:triggerName withAttributes:attributes];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received bad input in call to triggerInAppMessage; the first argument must be a non-empty String.");
     }
 }
 
 - (void)triggerInAppMessagesForSessionStart:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"triggerInAppMessagesForSessionStart" withCommand:command];
     [Localytics triggerInAppMessagesForSessionStart];
 }
 
 - (void)dismissCurrentInAppMessage:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"dismissCurrentInAppMessage" withCommand:command];
     [Localytics dismissCurrentInAppMessage];
 }
 
 - (void)setInAppMessageConfiguration:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"setInAppMessageConfiguration" withCommand:command];
     if (self.messagingDelegate) {
         self.messagingDelegate.inAppConfig = [command argumentAtIndex:0];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received call to setInAppMessageConfiguration but no messaging delegate is initialized.");
```

Comment:
Should we just init the messaging delegate here? Similar to how we do in React Native

Reply:
Considering it, I'll look into it today.

Reply:
This is tricky because of architecture differences - react uses an event emitter to do this and doesn't ever require a call to `setMessagingListener` since you can just register for certain events with `NativeAppEventEmitter.addListener(...)`.  

Cordova handles connecting to JS differently, it requires that we hold onto the call that came from JS so we can respond to it with results (and in the case of our listeners, respond to them a bunch of times).  For reference, look at `commandDelegate` on `CDMessagingDelegate`.

I think we could do this so that the configuration would work, but none of the `diy` options would work.  I think that is confusing, and would lead to most customers never setting the messaging listener but still trying to do the `diy` option.

Alternatively, we could opt to only error/log when the `diy` option is included but no messaging delegate is set. Maybe that is the best of both worlds? (but somehow also feels inconsistent).

The final consideration, is that is a change in behavior, which is fine for us to do, but may be confusing to different customers. 

Whatcha think?

Reply:
@Mriegel I'm OK with the behavior as it is, thanks for the explanation